node/src/database/rusqlite_wrappers.rs

```diff
@@ -1,58 +1,52 @@
-// unwinding.
+// The truth is the use of an explicit, object-like, transaction from 'rusqlite' is rare for our code.
+// That might be why we'd managed to live so long without being much constrained by absence of
+// the mock. While we did write some code using that kind of a transaction, we always acknowledged
+// only that we had to make and exception and leave that piece of code untested. The problem
+// with a mocked transaction has been that we often need to combine it with also a mocked connection.
+// Then the transaction springs from the connection. The upcoming nontrivial issues with lifetimes
+// have been briefly mentioned.
 //
-// The interface have three methods of which the one causing troubles is 'prepare'. It returns
-// `Statement`, a 'rusqlite' original structure always referenced back to the connection.
+// The interface a transaction needs have three methods of which 'prepare' causes the troubles.
+// It returns `Statement`, a 'rusqlite' structure that always keeps a reference to the connection.
 //
-// There isn't any vision of how we could put an end to using this method, despite how much tempting
-// it is. In theory, we could replace it fully by the existing 'execute' method, able to circumvent
-// any places where we are forced to manipulate with the 'Statement' because 'execute' returns only
-// simple data structures, instead, easily to be stored in a mock. Regardless the boldness of this
-// plan, new design issues, from smaller to larger, are anticipated.
+// There is now no decent vision of how to stop using that method, despite how much temptation it
+// provides. In theory, we could replace it by the existing 'execute' method, able to circumvent
+// any places where we now need to manipulate with the 'Statement', that is because 'execute'
+// returns, in contrary, only simple data structures, easy to be stored in a mock. Going that way,
+// other issues may be anticipated.
 //
-// One may think of a potential solution if another class existed to wrap the `Statement` up, that
-// we've said about that it causes these serious troubles. Also that's been tested out but we've
-// met insurmountable obstacles, whereupon the steeply increasing difficulty ended the efforts.
+// One may think about creating another class to wrap up the `Statement` to have a better control.
+// Tested, but its interface requires even more methods and so obstacles, whereupon the steeply
+// increasing difficulty ended the efforts.
 //
-// It is fair to admit our mock is poor at elegance and simplicity. Especially when it takes more
-// complex tests. However, compensation for that needs to be seen in the good portion of
-// possibilities it opens up for advance, thorough testing in those rare places where we do rely on
-// an explicit 'rusqlite' transaction and also let's keep in mind it used to seem undoable not so
-// far ago.
+// Fair to admit the mock is poor at elegance and simplicity. The more the more complex tests it
```

Comment:
Let's rephrase it to say, "Fair to admit that this mock is a gift to our codebase and it opens up the possibilities for thorough testing in those rare places..." (you may continue your paragraph from here)

:wink:

Reply:
I was thinking that you probably had been just joking, but the I also corrected this paragraph, hopefully you'll be okay with it.